src/tracks/playabletrack/wavetrack/ui/WaveTrackControls.cpp

```diff
@@ -760,26 +760,21 @@ void WaveTrackMenuTable::OnMergeStereo(wxCommandEvent &)
 
    const auto checkAligned = [](const WaveTrack& left, const WaveTrack& right)
    {
-      const auto isPlainInterval = [](const WaveTrack::Interval& interval)
-      {
-         return interval.GetTrimLeft() == .0 &&
-            interval.GetTrimRight() == .0 &&
-            interval.GetStretchRatio() == 1.0;
-      };
+      const auto eps = 0.5 / left.GetRate();
       const auto rightIntervals = right.Intervals();
       for(const auto& a : left.Intervals())
       {
-         if(!isPlainInterval(*a))
-            return false;
-
          auto it = std::find_if(
             rightIntervals.begin(),
             rightIntervals.end(),
             [&](const auto& b)
             {
-               return a->Start() == b->Start() &&
-                  a->End() == b->End() &&
-                  isPlainInterval(*b);
+               //Start() and End() are always snapped to a sample grid
+               return std::fabs(a->Start() - b->Start()) < eps &&
+                  std::fabs(a->End() - b->End()) < eps &&
+                  a->GetTrimLeft() == b->GetTrimLeft() &&
```

Comment:
I wonder if we need the same `eps` here?

Reply:
Or, probably `numeric_limits<double>::epsilon() * std::max(a->GetTrimRight(), b->GetTrimRight())`

Reply:
Yes, I think it could be `epsilon * max`, but not `0.5 * rate` because trim are no longer aligned with sample grid. Thanks